src/core/src/sys/interactions.rs

```diff
@@ -443,4 +445,17 @@ mod test {
         interactions.add_global(Box::new(Wolf::new(1.0)));
         assert_eq!(interactions.globals().len(), 1);
     }
+
+    #[test]
+    fn test_globals_cutoff() {
```

Comment:
+1 for making it part of `Interaction` because this is very likely to be reused later.